mmds/src/data_store.rs

```diff
@@ -1,27 +1,31 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::fmt;
+
 use serde_json::Value;
 
 /// The Mmds is the Microvm Metadata Service represented as an untyped json.
-#[derive(Clone)]
 pub struct Mmds {
     data_store: Value,
     is_initialized: bool,
 }
 
+/// Errors thrown while accessing and configuring the MMDS data store.
 #[derive(Debug, PartialEq)]
 pub enum Error {
+    /// The MMDS resource does not exist.
     NotFound,
+    /// Failure while adding non-strings values to the MMDS data-store.
     UnsupportedValueType,
 }
 
-impl Error {
-    pub fn to_string(&self) -> String {
+impl fmt::Display for Error {
```

Comment:
nit: This change isn't related to the documentation, can you add it to a different commit?